microcloud/cmd/microcloud/ask.go

```diff
@@ -816,3 +817,44 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 
 	return nil
 }
+
+// askClustered checks whether any of the selected systems have already initialized any expected services.
+// If a service is already initialized on some systems, we will offer to add the remaining systems, or skip that service.
+// If multiple systems have separately initialized the same service, we will abort initialization.
+// Preseed yamls will have a flag that sets whether to reuse the cluster.
+// In auto setup, we will expect no initialized services so that we can be opinionated about how we configure the cluster without user input.
+func (c *CmdControl) askClustered(s *service.Handler, autoSetup bool, systems map[string]InitSystem) error {
+	expectedServices := make(map[types.ServiceType]service.Service, len(s.Services))
+	for k, v := range s.Services {
+		expectedServices[k] = v
+	}
+
+	for serviceType := range expectedServices {
+		initializedSystem, err := checkClustered(s, autoSetup, serviceType, systems)
+		if err != nil {
+			return err
+		}
+
+		if initializedSystem != "" {
+			question := fmt.Sprintf("%q is already part of a %s cluster. Use this cluster to MicroCloud, or skip %s? (reuse/skip) [default=reuse]", initializedSystem, serviceType, serviceType)
```

Comment:
The wording here is a bit off. Perhaps:
```golang
fmt.Sprintf("%q is already part of a %s cluster. Do you want to add this cluster to Microcloud? (add/skip) [default=add]", initializedSystem, serviceType)
```

Also, do we need to capitalise or otherwise format the `serviceType` string?

Reply:
No, the type represents consrs which are actual service names, capitalized appropriately. So `MicroCloud`, `MicroOVN`, `MicroCeph`, `LXD`.